sql/rename.go

```diff
@@ -136,3 +136,57 @@ func (p *planner) RenameTable(n *parser.RenameTable) (planNode, error) {
 
 	return &valuesNode{}, nil
 }
+
+// RenameIndex renames the index.
+// Privileges: WRITE on database.
+//   Notes: postgres requires the table owner.
+//          mysql requires ALTER, CREATE, and INSERT on the table.
+func (p *planner) RenameIndex(n *parser.RenameIndex) (planNode, error) {
+	if n.NewName == "" {
+		return nil, errEmptyIndexName
+	}
+
+	if err := n.Name.NormalizeTableName(p.session.Database); err != nil {
+		return nil, err
+	}
+
+	if n.Name.Index() == string(n.NewName) {
+		// Noop.
```

Comment:
we had a similar question on #2101 and ended up going with noop in this case.
It might be more informative to check for object existence first.
I don't mind too much either way as long as we're consistent.